src/dotnet/CarbonAware.DataSources.Json/JsonDataSource.cs

```diff
@@ -67,9 +68,15 @@ private IEnumerable<EmissionsData> FilterByDateRange(IEnumerable<EmissionsData>
     {
         DateTimeOffset end;
         DateTimeOffset.TryParse(endDate.ToString(), out end);
-        data = data.Where(ed => ed.TimeBetween(startDate, end));  
-
-        return data;
+        var (newStartTime, newEndTime) = IntervalHelper.ShiftDate(startDate, end, MinSamplingWindow);
+        var newWindowData = data.Where(ed => ed.TimeBetween(newStartTime.DateTime, newEndTime));
```

Comment:
Is there a reason why we are converting `newStartTime` to a `DateTime` here?

Reply:
hmm, leftover from previous commits. Good catch.